exercises/luhn/canonical-data.json

```diff
@@ -29,6 +29,46 @@
       "description": "valid strings with a non-digit added become invalid",
       "input": "046a 454 286",
       "expected": false
+    },
+    {
+      "description": "punctuation is not allowed",
+      "input": "055-444-285",
+      "expected": false
+    },
+    {
+      "description": "symbols are not allowed",
+      "input": "055£ 444$ 285",
+      "expected": false
+    },
+    {
+      "description": "single zero with space is invalid",
+      "input": " 0",
+      "expected": false
+    },
+    {
+      "description": "lots of zeros are invalid",
+      "input": " 00000",
+      "expected": false
+    },
+    {
+      "description": "another valid sin",
+      "input": "055 444 285",
+      "expected": true
+    },
+    {
+      "description": "nine doubled is nine",
+      "input": "091",
+      "expected": true
+    },
+    {
+      "description": "must double from right",
+      "input": "0091",
+      "expected": true
+    },
+    {
+      "description": "simple valid sin",
```

Comment:
also another note that this is another case that will reveal doubling from the wrong side - if doubling the wrong digits here we will get 14 instead of 10 and falsely think this is invalid.

This could possibly make the "must double from right" case redundant.